supermarq-benchmarks/supermarq/qcvv/base_experiment.py

```diff
@@ -502,36 +568,70 @@ def _canonicalize_bitstring(key: int | str, num_qubits: int) -> str:
 
     @staticmethod
     def _canonicalize_probabilities(
-        probabilities: dict[str | int, float], num_qubits: int
+        results: dict[str, int] | dict[str, float] | dict[int, int] | dict[int, float],
```

Comment:
this should work:
```suggestion
        results: Mapping[str, float] | Mapping[int, float],
```
(and also obviate the type: ignore below)

Reply:
Thanks! I had tried using a mapping but couldn't find quite the right way!